dowhy/causal_estimators/linear_regression_estimator.py

```diff
@@ -35,20 +31,38 @@ def __init__(self, *args, **kwargs):
         self.logger.info(self.symbolic_estimator)
         self._linear_model = None
 
-    def _estimate_effect(self):
-        features, self._linear_model = self._build_linear_model()
-        coefficients = self._linear_model.params[1:] # first coefficient is the intercept
-        self.logger.debug("Coefficients of the fitted linear model: " +
+    def _estimate_effect(self, data_df=None, need_conditional_estimates=None):
+        # TODO make treatment_vale and control value also as local parameters
+        if data_df is None:
+            data_df = self._data
+        if need_conditional_estimates is None:
+            need_conditional_estimates = bool(self._effect_modifier_names)
+        # Checking if the model is already trained
+        if not self._linear_model:
```

Comment:
It will make finding the value of the linear estimate more efficient. However, would it not be better if we first check if there has been a change in the variables of interest ?

Reply:
The expected use is that the user accesses this from the estimate object, so we do not allow changes to the parameters of the model once fit. I've added a warning to make this clear.